### DISCUSSION TITLE
order by strange result on nullable

### DISCUSSION THREAD

**Discussion by minyoung90** [sqlalchemy/sqlalchemy#8932] [Usage Questions] [answered]
Recently, I had lots of data, and I have pagination API on it.

```python
# after complicated conditions on "stmt"
# e.g. order == desc(target_column)
# e.g. limit == 10
# e.g. offset == 0
stmt = stmt.order_by(order)
stmt = stmt.limit(limit).offset(offset)
result = orm_session.execute(stmt).scalars().unique()
```

Above is a snap for pagination API code.
It is fine when paginate on not null column, but when it is paginated on nullable I can see strange result.

What I saw was, some of rows are disappeared or overlapped in results.
For example, id 23 appears in offset 0, and id 23 reappears in offset 1. In addition, there were IDs that did not appear in all offsets. In all of the above cases, all conditions are the same, and if only the not null column is changed from nullable, all data is accurately and well output.

Is there anything I did wrong on the nullable column? (What I did was to use asc() or desc() only with order by)

**Comment by CaselIT** [comment, marked as answer]
Hi,

This is more of an SQL question, not really a SQLAlchemy one.
If you have duplicated values in the order by column I think the db is free to order the rows as it pleases, so it may happen that the same row is returned in different positions in different queries.

If you want a stable ordering on a duplicate column you can add an unique column to the order by clause

**Reply by minyoung90** [reply]
Thank you!
